### PR TITLE
Apply V3 Design / Link Social - Issue/v3-2134

### PR DESCRIPTION
**Pull request by o-hook**
### Pull Request Checklist

- [x] I have read and accept the [Contributor License Agreement](https://github.com/porscheofficial/oss-docs/blob/67c5b24838a293ce7a964884e6005eb71f2b8579/CONTRIBUTOR_LICENSE_AGREEMENT.md)

#### References

- Preview: https://designsystem.porsche.com/issue/v3-2134/components/link-social/examples

#### Scope

Short description of the pull request scope

### CODE REVIEW

File: packages/components-js/tests/vrt/puppeteer/specs/link-social.vrt.ts [hunk 1]
```diff
@@ -27,7 +27,6 @@ xit('should have no visual regression for :hover + :focus-visible', async () =>
 
       const getElementsMarkup: GetThemedMarkup = (theme) => `
         <p-link-social theme="${theme}" href="https://www.porsche.com/">Fallback</p-link-social>
-        <p-link-social theme="${theme}" href="https://www.porsche.com/" hide-label="true">Fallback</p-link-social>
```
Comment (mhaegeleMHP) [inline packages/components-js/tests/vrt/puppeteer/specs/link-social.vrt.ts:30]: Why is this removed?

File: packages/components/src/components/link-social/link-social.tsx [hunk 2]
```diff
@@ -51,11 +52,23 @@ export class LinkSocial {
 
   public componentWillLoad(): void {
     throwIfInvalidLinkUsage(this.host, this.href);
+    console.warn(
```
Comment (mhaegeleMHP) [inline packages/components/src/components/link-social/link-social.tsx:55]: Iam not sure, but if there are more than one component that get deprecated id could make sense to create a helper for this and reuse to ensure the warning is everywhere the same?

File: packages/components-js/tests/vrt/puppeteer/specs/link-social.vrt.ts (hunk 1 above)
Comment (o-hook) [reply packages/components-js/tests/vrt/puppeteer/specs/link-social.vrt.ts]: It's there aigain.

File: packages/components/src/components/link-social/link-social.tsx (hunk 2 above)
Comment (o-hook) [reply packages/components/src/components/link-social/link-social.tsx]: Helper function was created.

File: packages/components/src/utils/validation/throwIfComponentIsDeprecated.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import { getTagName } from '../../utils';
+
+export const throwIfComponentIsDeprecated = (host: HTMLElement, message: string): void => {
```
Comment (denyo) [inline packages/components/src/utils/validation/throwIfComponentIsDeprecated.ts:3]: That's not throwing anything 🙈

File: packages/components/src/components/link-social/link-social.tsx [hunk 4]
```diff
@@ -51,11 +53,24 @@ export class LinkSocial {
 
   public componentWillLoad(): void {
     throwIfInvalidLinkUsage(this.host, this.href);
+    throwIfComponentIsDeprecated(
```
Comment (denyo) [inline packages/components/src/components/link-social/link-social.tsx:56]: A generic util would take the new component name as the 2nd parameter and the message would be part of the util.

File: packages/components/src/utils/validation/throwIfComponentIsDeprecated.ts (hunk 3 above)
Comment (o-hook) [reply packages/components/src/utils/validation/throwIfComponentIsDeprecated.ts:3]: What would you suggest as function name? `warnIfCOmponentIsDeprecated`?

File: packages/components/src/components/link-social/link-social.tsx (hunk 4 above)
Comment (o-hook) [reply packages/components/src/components/link-social/link-social.tsx:56]: Yeah, but there is not always a "new" component. SOmetimes a component is just "deprecated" and nothing else. So we need a possibility of customization in the warning message.

**Review by denyo — APPROVED** [review]
